### PR TITLE
Adding configuration to set the streaming request timeout

### PR DESCRIPTION
**Pull request by htrivedi99**
## :rocket: What

A user was facing issues seeing 4xx status code after a 60 second timeout. They wanted to make this parameter configurable so that they can increase the timeout for streaming use-cases.

## :computer: How

Inside the `config.yaml` under runtime, a new key can be set called `streaming_read_timeout`. Setting this key overrides the default streaming timeout which is 60 seconds. This way, users can configure the streaming timeout through the configuration file for one-off use-cases and we don't have to change the default timeout for the rest of the users.

## :microscope: Testing

Tested it locally with a higher timeout and lower timeout to ensure it fails on the lower timeout and succeeds on the higher timeout.

### CODE REVIEW

**Review by bolasim — COMMENTED** [review]
can you please add some integration tests for this?

**Comment by htrivedi99** [comment]
> can you please add some integration tests for this?

bet

File: truss/templates/server/model_wrapper.py [hunk 1]
```diff
@@ -332,7 +336,7 @@ async def _response_generator():
                     while True:
                         chunk = await asyncio.wait_for(
                             response_queue.get(),
-                            timeout=STREAMING_RESPONSE_QUEUE_READ_TIMEOUT_SECS,
+                            timeout=streaming_read_timeout,
```
Comment (squidarth) [inline truss/templates/server/model_wrapper.py:339]: could you add a comment here about what exactly this timeout refers to? (ie: the gap between responses in a stream must be < streaming_read_timeout

**Review by squidarth — COMMENTED** [review]
lgtm, +1 to bola's comment about integration test

File: truss/tests/templates/server/test_model_wrapper.py [hunk 2]
```diff
@@ -68,3 +68,20 @@ def test_model_wrapper_load_error_more_than_allowed(app_path, helpers):
         # Allow load thread to execute
         time.sleep(1)
         assert model_wrapper.load_failed()
+
+
+@pytest.mark.integration
+async def test_model_wrapper_streaming_timeout(app_path):
```
Comment (squidarth) [inline truss/tests/templates/server/test_model_wrapper.py:74]: it just feels like we're testing whether we're passing the config to the model_wrapper correctly. Could you add a test to these tests: https://github.com/basetenlabs/truss/blob/main/truss/tests/test_model_inference.py.

I'm imagining a test where we:
* set the streaming_read_timeout to something low (like a second)
* in the truss have it stream, but wait 2 seconds between yielding comment
* Ensure that a timeout error is logged

File: truss/test_data/test_streaming_read_timeout/config.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+apply_library_patches: true
```
Comment (squidarth) [inline truss/test_data/test_streaming_read_timeout/config.yaml:1]: could you just include the fields that we actually need?

File: truss/tests/test_model_inference.py [hunk 4]
```diff
@@ -232,6 +232,27 @@ def test_async_streaming():
...
+        truss_root = Path(__file__).parent.parent.parent.resolve() / "truss"
+
+        truss_dir = truss_root / "test_data" / "test_streaming_read_timeout"
+
+        tr = TrussHandle(truss_dir)
+
+        _ = tr.docker_run(local_port=8090, detach=True, wait_for_server_ready=True)
+        truss_server_addr = "http://localhost:8090"
+        predict_url = f"{truss_server_addr}/v1/models/model:predict"
+
```
Comment (squidarth) [inline truss/tests/test_model_inference.py:247]: would also be good to assert that this shows up as a timeout error in the logs (there are other reasons a stream might get terminated) -- could you use one of our helpers to do this? See https://github.com/basetenlabs/truss/blob/main/truss/tests/test_model_inference.py#L384 for an example

**Pull request merged**